### PR TITLE
[Fixes #69] Use nlpsandboxclient

### LINKED ISSUE #69: Rely on the nlpsandbox-client
The Deidentifier should not just fail and give a 500 if it can't get a good response from one or more of the annotators

### PR DESCRIPTION
**Pull request by cascadianblue**
Also addresses #87

### CODE REVIEW

**Comment by tschaffter** [comment]
Adding @thomasyu888 as a required reviewer for this ticket

**Review by tschaffter — APPROVED** [review]
Looks good to me!

File: server/openapi_server/controllers/tool_controller.py [hunk 1]
```diff
@@ -33,4 +35,12 @@ def get_tool_dependencies():  # noqa: E501
...
     """
-    return annotators.get_annotators_info(), 200
+    config = Config()
+    tool_dependencies = []
+    for hostname in (
+        config.date_annotator_api_url,
+        config.person_name_annotator_api_url,
+        config.physical_address_annotator_api_url
+    ):
+        tool_dependencies.append(client.get_tool(host=hostname).to_dict())
```
Comment (thomasyu888) [inline server/openapi_server/controllers/tool_controller.py:45]: This only gets the tool, I realize there's actually a toolDependency endpoint, were you looking for that instead?

File: server/requirements.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
 connexion==2.7.0
 mongoengine==0.22.1
+nlpsandbox-client==0.3.1
```
Comment (thomasyu888) [inline server/requirements.txt:3]: Bump to 0.4.1

**Review by thomasyu888 — CHANGES_REQUESTED** [review]

File: server/openapi_server/controllers/tool_controller.py (hunk 1 above)
Comment (cascadianblue) [reply server/openapi_server/controllers/tool_controller.py:45]: Hmmm, so the PHI Deidentifier should get present a flattened list of both its dependencies, and its dependencies' dependencies?

I suppose that makes sense if we are allowing annotator docker images to be dependent on other docker images. Is that so @tschaffter ?

File: server/requirements.txt (hunk 2 above)
Comment (cascadianblue) [reply server/requirements.txt]: Bumped!

File: server/openapi_server/controllers/tool_controller.py (hunk 1 above)
Comment (cascadianblue) [reply server/openapi_server/controllers/tool_controller.py]: I'm moving this conversation to issue #94

**Pull request merged**